modules/claims_api/app/sidekiq/claims_api/poa_vbms_updater.rb

```diff
@@ -51,5 +46,41 @@ def perform(power_of_attorney_id) # rubocop:disable Metrics/MethodLength
     def allow_address_change?(poa_form)
       poa_form.form_data['consentAddressChange']
     end
+
+    def update_poa_access(poa_form:, participant_id:, poa_code:, allow_poa_access:)
+      # allow_poa_c_add reports 'No Data' if sent lowercase
+      if Flipper.enabled? :claims_api_poa_vbms_updater_uses_local_bgs
+        service = corporate_update_service
+        response = service.update_poa_access(
+          participant_id:,
+          poa_code:,
+          allow_poa_access:,
+          allow_poa_c_add: allow_address_change?(poa_form) ? 'Y' : 'N'
+        )
+      else
+        service = bgs_ext_service
+        response = service.corporate_update.update_poa_access(
+          participant_id:,
+          poa_code:,
+          allow_poa_access:,
+          allow_poa_c_add: allow_address_change?(poa_form) ? 'Y' : 'N'
+        )
+      end
+      response
```

Comment:
If we remove

```rb
def corporate_update
  self
end
```

from `CorporateUpdateWebService`, we should be able to consolidate this logic:

```suggestion
      service = if Flipper.enabled? :claims_api_poa_vbms_updater_uses_local_bgs
                  corporate_update_service
                else
                  bgs_ext_service
                end
        
      service.update_poa_access(
        participant_id:,
        poa_code:,
        allow_poa_access:,
        allow_poa_c_add: allow_address_change?(poa_form) ? 'Y' : 'N'
      )
```

Reply:
@tycol7  Wouldn't the call to the older bgs servce error out there since, I _believe_ at least, it requires the chained `.corporate_access.` in it?  The old method call being `service.corporate_update.update_poa_access` vs. `service.update_poa_access` ?

Reply:
I see now, `corporate_update` applies to the _older_ service. So it _kind of_ makes sense to have the [`corporate_update`](https://github.com/department-of-veterans-affairs/vets-api/pull/19547#discussion_r1861043056) on the new service. If we have `corporate_update` on the old and new service, we should be able to use the [above logic](https://github.com/department-of-veterans-affairs/vets-api/pull/19547#discussion_r1861048943) to call `service.corporate_update.update_poa_access`.

That said, I would prefer we _not_ do that and remove the `corporate_update` from the new service, since it does not fit with the pattern of our new services. We will remove this if/else logic with the feature flag anyway, so it does not have to be pretty.

Reply:
Removed the `corporate_update` method but retained the original logic so the call to the old service with `service.corporate_update.update_poa_access` still succeeds.